pkg/provider/kubernetes/cluster.go

```diff
@@ -269,13 +269,19 @@ func (p *ClusterProvider) Get(ctx context.Context, userInfo *provider.UserInfo,
 	if options == nil {
 		options = &provider.ClusterGetOptions{}
 	}
-	seedImpersonatedClient, err := createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
-	if err != nil {
-		return nil, err
+
+	client := p.client
+
+	if !userInfo.IsAdmin {
+		var err error
+		client, err = createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I have to admit, I do not fully understand what this code is doing. 

As I understand it, if the user is not an admin we overrwrite the client with one that is impersonated; otherwise you just get the normal client, which I assume has permissions to get all clusters?)

So would it not be better if `createImpersonationClientWrapperFromUserInfo` checks for if someone is admin and if so gives back and impersonation to do anything?

Reply:
> As I understand it, if the user is not an admin we overrwrite the client with one that is impersonated; otherwise you just get the normal client, which I assume has permissions to get all clusters?)

That's correct.

In my understanding the client that we assign here for admins (`p.client`) is the privileged one that is configured at the start of API - it's a master's admin token. It has nothing to do with impersonalisation as it is not assigned to any specific person. Impersonate client is created based on the data in OIDC token which is later used to map privileges in role bindings.

Regarding `createImpersonationClientWrapperFromUserInfo` function, it may not be simple to return impersonated client with all required permissions.

Reply:
I see now.